lib/mongo/client.rb

```diff
@@ -1311,11 +1319,20 @@ def valid_compressors(compressors)
                        "This compressor will not be used.")
           false
         else
+
           true
         end
       end
     end
 
+    def validate_snappy_compression!
+      require 'snappy'
+    rescue LoadError
+      raise LoadError, "Cannot enable snappy compression because the snappy gem " \
```

Comment:
This should forward the original exception message via the `#{e.class}: #{e}` pattern and I think it would be better to raise an exception derived from Mongo::Error, if the driver is raising one.